SQLQuery_LibraryBooksBD.sql

```diff
@@ -0,0 +1,842 @@
+(
+	[UserId] ASC,
+	[BookId] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, SORT_IN_TEMPDB = OFF, IGNORE_DUP_KEY = OFF, ONLINE = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
+GO
+/****** Object:  Index [IX_UsersAndRoles]    Script Date: 19.07.2021 19:13:04 ******/
+ALTER TABLE [dbo].[UsersAndRoles] ADD  CONSTRAINT [IX_UsersAndRoles] UNIQUE NONCLUSTERED 
+(
+	[UserId] ASC,
+	[RoleId] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, SORT_IN_TEMPDB = OFF, IGNORE_DUP_KEY = OFF, ONLINE = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
+GO
+ALTER TABLE [dbo].[AssignedRoles]  WITH CHECK ADD  CONSTRAINT [FK_AssignedRoles_ToAuthData] FOREIGN KEY([AuthId])
+REFERENCES [dbo].[AuthData] ([Id])
+ON UPDATE CASCADE
+ON DELETE CASCADE
+GO
+ALTER TABLE [dbo].[AssignedRoles] CHECK CONSTRAINT [FK_AssignedRoles_ToAuthData]
+GO
+ALTER TABLE [dbo].[AssignedRoles]  WITH CHECK ADD  CONSTRAINT [FK_AssignedRoles_ToRole] FOREIGN KEY([RoleId])
+REFERENCES [dbo].[Role] ([Id])
+ON UPDATE CASCADE
+ON DELETE CASCADE
+GO
+ALTER TABLE [dbo].[AssignedRoles] CHECK CONSTRAINT [FK_AssignedRoles_ToRole]
+GO
+ALTER TABLE [dbo].[Book]  WITH CHECK ADD  CONSTRAINT [FK_Book_ToAuthor] FOREIGN KEY([AuthorId])
+REFERENCES [dbo].[Author] ([Id])
+ON UPDATE CASCADE
+ON DELETE SET NULL
+GO
+ALTER TABLE [dbo].[Book] CHECK CONSTRAINT [FK_Book_ToAuthor]
+GO
+ALTER TABLE [dbo].[UserInfo]  WITH CHECK ADD  CONSTRAINT [FK_UserInfo_ToAuthData] FOREIGN KEY([UserId])
+REFERENCES [dbo].[AuthData] ([Id])
+ON UPDATE CASCADE
+ON DELETE CASCADE
+GO
+ALTER TABLE [dbo].[UserInfo] CHECK CONSTRAINT [FK_UserInfo_ToAuthData]
+GO
+ALTER TABLE [dbo].[UsersAndBooks]  WITH CHECK ADD  CONSTRAINT [FK_UsersAndBooks_ToAuthData] FOREIGN KEY([UserId])
+REFERENCES [dbo].[AuthData] ([Id])
+ON UPDATE CASCADE
+ON DELETE CASCADE
+GO
+ALTER TABLE [dbo].[UsersAndBooks] CHECK CONSTRAINT [FK_UsersAndBooks_ToAuthData]
+GO
+ALTER TABLE [dbo].[UsersAndBooks]  WITH CHECK ADD  CONSTRAINT [FK_UsersAndBooks_ToBook] FOREIGN KEY([BookId])
+REFERENCES [dbo].[Book] ([Id])
+ON UPDATE CASCADE
+ON DELETE CASCADE
+GO
+ALTER TABLE [dbo].[UsersAndBooks] CHECK CONSTRAINT [FK_UsersAndBooks_ToBook]
+GO
+/****** Object:  StoredProcedure [dbo].[AuthData_Create]    Script Date: 19.07.2021 19:13:04 ******/
+SET ANSI_NULLS ON
+GO
+SET QUOTED_IDENTIFIER ON
+GO
+CREATE PROCEDURE [dbo].[AuthData_Create]
+	@Login nvarchar(50),
+	@HashPass nvarchar(MAX),
+	@SurName nvarchar(50),
+	@FirstName nvarchar(50),
+	@SecondName nvarchar(50),
+	@DateOfBirth datetime,
+	@Phone nvarchar(50)
+
+AS
+BEGIN
+    INSERT INTO AuthData(Login, HashPass)
+	VALUES(@Login, @HashPass)
+END
+BEGIN
+	INSERT INTO UserInfo(UserId, SurName, FirstName, SecondName, DateOfBirth, [Phone])
+	VALUES((SELECT MAX(Id) FROM AuthData), @SurName, @FirstName, @SecondName, @DateOfBirth, @Phone)
+END
+BEGIN
+	INSERT INTO UsersAndRoles(UserId, RoleId)
+	VALUES((SELECT MAX(Id) FROM AuthData), 1)
+END
```

Comment:
You don't need `BEGIN..END` blocks here it's for IF, WHILE blocks, etc.. A simple `;` would be enough

Also - at least leave a comment on what is 1

Also, Id for last inserted AuthData (which uses IDENTITY - autoincrement) can be retrieved by SCOPE_IDENTITY() function - no need for selecting MAX